### PR TITLE
COMMERCE-12929 - Guest provided address saved and cached in and authenticated user is offered to use it

### PR DESCRIPTION
**Pull request by jaydawu**
https://liferay.atlassian.net/browse/COMMERCE-12929
Resent from https://github.com/liferay-commerce/liferay-portal/pull/4483

### CODE REVIEW

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/GuestCheckout.testcase [hunk 1]
```diff
@@ -695,6 +695,150 @@ definition {
 		}
 	}
 
+	@description = "COMMERCE-12929. As an authenticated user, I want to be able to view the open order address for checkout that is not inherited from the guest's address and no default address pre-selected"
+	@priority = 5
+	test CannotGuestAddressBeInheritedForLoggedUser {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/GuestCheckout.testcase:700]:
```suggestion
	test CanLoggedUserNotInheritGuestAddress {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/GuestCheckout.testcase [hunk 2]
```diff
@@ -695,6 +695,150 @@ definition {
...
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/minium/catalog");
+
+			FrontStore.addInitializerProductToCart(
+				productName = "U-Joint",
+				productQuantity = 1);
+		}
+
+		task ("And adds an address in the checkout page and moves to next step") {
+			CommerceCheckout.gotoCheckoutViaMiniCart();
+
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/GuestCheckout.testcase:791]: Based on the ticket's description, we can see that the list of available addresses is either disabled/not showing any address. Could we add a check for this at this point? Thanks!

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/GuestCheckout.testcase (hunk 2 above)
Comment (jaydawu) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/GuestCheckout.testcase:791]: Hi @lorenzolapescara, the address list selector doesn't show for the guest user, so I just added an assertion for it as the selector is not present, thanks!
result

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:relevant

**Review by lorenzolapescara — APPROVED** [review]

**Comment by lorenzolapescara** [comment]
Failures in the commerce-functional suite are not related.
The relevant suite is green.
Forwarding.

**Comment by lorenzolapescara** [comment]
ci:forward

**Pull request closed without merging**